### DISCUSSION TITLE
Collections: Include transformKeys method

### DISCUSSION THREAD

**Discussion by levu42** [laravel/framework#38269] [Ideas] [answered]
I would like to see a method that transforms keys, e.g. for converting them all to lower case (in case they're strings) or multiplying them by 2 (when it's an array).

Example:

```php
$collection = collect([1, 2, 3]);
$collection->transformKeys(function($key) {
  return $key * 2;
});

// [0 => 1, 2 => 2, 4 => 3]

$collection = collect([
  'Content-Type' => 'application/json',
  'X-MyApp-Access-Token' => 'abc123'
 ]);

$collection->transformKeys(function($key) {
  return strtolower($key);
});

// [
//   'content-type' => 'application/json'
//   'x-myapp-access-token' => 'abc123'
// ]
```

Currently I'm doing it by using $collection->flip()->transform(...)->flip() but that's not working for when two elements have different keys but the same value.

**Comment by dennisprudlo** [comment, marked as answer]
I would usually use `mapWithKeys` in such a case:

```php
$headers = collect([
    'Content-Type' => 'application/json',
    'X-MyApp-Access-Token' => 'abc123'
]);

$headers = $headers->mapWithKeys(fn ($value, $key) => [ strtolower($key) => $value ]);
```

**Reply by levu42** [reply]
Oh, great to know that this works! It's not mentioned in the documentation that the callback function has a 2nd argument $key. Thanks a lot!
